annotator/ffa/unknownbinaries/filterknownbinaries.go

```diff
@@ -0,0 +1,107 @@
+// Package unknownbinaries removes all packages extracted by unknown binaries
+// filters out the known binaries, and records the remaining as a finding.
+package unknownbinaries
+
+import (
+	"context"
+	"fmt"
+	"maps"
+	"slices"
+	"strings"
+
+	"github.com/google/osv-scalibr/annotator"
+	"github.com/google/osv-scalibr/annotator/ffa/unknownbinaries/internal/dpkgfilter"
+	"github.com/google/osv-scalibr/annotator/ffa/unknownbinaries/internal/filter"
+	"github.com/google/osv-scalibr/extractor"
+	"github.com/google/osv-scalibr/extractor/filesystem/ffa/unknownbinary"
+	"github.com/google/osv-scalibr/inventory"
+	"github.com/google/osv-scalibr/plugin"
+)
+
+// Name of the plugin
+const Name = "ffa/unknownbinaries"
+
+// List of filters to apply to exclude known binaries
+var filters = []filter.Filter{
+	dpkgfilter.DpkgFilter{},
+}
+
+// UnknownBinaryAnnotator further processes the UnknownBinaryExtractor
+type UnknownBinaryAnnotator struct {
+}
+
+// Annotate removes all packages extracted by unknown binaries,
+// filters out the known binaries, and records the remaining as a finding.
+func (enr UnknownBinaryAnnotator) Annotate(ctx context.Context, input *annotator.ScanInput, inv *inventory.Inventory) error {
+	unknownBinariesSet := map[string]struct{}{}
+	filteredPackages := make([]*extractor.Package, 0, len(inv.Packages))
+	for _, e := range inv.Packages {
+		//Plugin contains unknownbinary.Name
+		if !slices.Contains(e.Plugins, unknownbinary.Name) {
+			filteredPackages = append(filteredPackages, e)
+			continue
+		}
+
+		unknownBinariesSet[e.Locations[0]] = struct{}{}
+	}
+
+	// Remove all unknown binary packages from output packages
+	inv.Packages = filteredPackages
+
+	// Two sets of filters, one with a hashset that gets elements deleted out of, and another with a simple loop
+
+	// Hash set filter
+	for _, f := range filters {
+		err := f.HashSetFilter(ctx, input.ScanRoot.FS, unknownBinariesSet)
+		if err != nil {
+			return fmt.Errorf("%s halted at %q (%q) because %w", enr.Name(), input.ScanRoot.Path, f.Name(), err)
+		}
+	}
+
+	remainingPaths := maps.Keys(unknownBinariesSet)
+	filteredRemainingPaths := make([]string, 0, len(unknownBinariesSet))
+
+	// Loop Filter
+RemainingPathsLoop:
+	for p := range remainingPaths {
+		for _, f := range filters {
+			if f.ShouldExclude(ctx, input.ScanRoot.FS, p) {
+				continue RemainingPathsLoop
+			}
+		}
+
+		filteredRemainingPaths = append(filteredRemainingPaths, p)
+	}
+
+	// Finally join the remaining unknown paths as a big finding.
+	findings := strings.Join(filteredRemainingPaths, "\n")
+	inv.GenericFindings = append(inv.GenericFindings, &inventory.GenericFinding{
```

Comment:
I still don't think we should convert packages to findings here since these aren't really security findings.

We should just keep the known binaries as package, or introduce a new Inventory type for them.